### ISSUE TITLE
[java] `EnumComparison` false positive for any object when object.equals(null)

### ISSUE THREAD

**Issue by ben-manes** [pmd/pmd#6291]
**Affects PMD Version:** 7.19.0

**Rule:** EnumComparison
<https://docs.pmd-code.org/pmd-doc-7.19.0/pmd_rules_java_bestpractices.html#enumcomparison>

**Description:**

**Code Sample demonstrating the issue:**
<img width="1060" height="190" alt="Image" src="https://github.com/user-attachments/assets/201437b9-fede-432c-9d51-e169168f847d" />

<img width="597" height="109" alt="Image" src="https://github.com/user-attachments/assets/0f43f650-ba00-43ec-994b-a57f9524c820" />

<img width="583" height="190" alt="Image" src="https://github.com/user-attachments/assets/b60b101c-9e71-48bb-8793-b9a1a32acc92" />

This is reproducible as,

```java
assertThat(new Object().equals(null)).isFalse();
```

**Expected outcome:**
Only enums should be reported on and null should not be assumed to be an enum.

**Running PMD through:** *Gradle*
